Add tools directory, md2swagger

bserdar opened the pull request.

dcrissman commented on tools/md2swagger.py:
```diff
@@ -0,0 +1,103 @@
+        properties['format']='int64'
+    elif type=='string':
+        properties['type']='string'
+    return properties;
```
how about a default case if unexpected type is passed in

dcrissman commented on tools/md2swagger.py:
```diff
@@ -0,0 +1,103 @@
+    elif s[-1]=='s':
+        return s[0:-1]
+    else:
+        return s;
```
Seems frail

dcrissman replied:
> Do you have a better idea to convert a plural to a singular?

In fairness, no. My thought was that a word that correctly ends with 's' would have problems, but the more I think about it, that may be a fringe case anyway.

Could you explain the need for this conversion?

dcrissman replied:
> Without this, we had UserLegalEntities as a classname.

I won't hold this up further on this point. This is certainly good enough and we can adjust if/when we find a use case. Thanks.

bserdar commented:
I'll add the default case for unknown type (string).

Do you have a better idea to convert a plural to a singular?

On Tue, Dec 6, 2016 at 7:43 PM, Dennis Crissman <notifications@github.com>
wrote:

> *@dcrissman* commented on this pull request.
> ------------------------------
>
> In tools/md2swagger.py
> <https://github.com/lightblue-platform/lightblue-core/pull/744#pullrequestreview-11726819>
> :
>
> > +    if type=='bigdecimal' or type=='biginteger':
> +        properties['type']='string'
> +    elif type=='boolean':
> +        properties['type']='boolean'
> +    elif type=='date':
> +        properties['type']='string'
> +        properties['format']='date-time'
> +    elif type=='double':
> +        properties['type']='number'
> +        properties['format']='double'
> +    elif type=='integer':
> +        properties['type']='integer'
> +        properties['format']='int64'
> +    elif type=='string':
> +        properties['type']='string'
> +    return properties;
>
> how about a default case if unexpected type is passed in
> ------------------------------
>
> In tools/md2swagger.py
> <https://github.com/lightblue-platform/lightblue-core/pull/744#pullrequestreview-11726819>
> :
>
> > +    elif type=='string':
> +        properties['type']='string'
> +    return properties;
> +
> +def capital1(s):
> +    return s[0].upper()+s[1:]
> +
> +def singular(s):
> +    if s[-3:]=='ies':
> +        return s[0:-3]+'y'
> +    elif s[-4:]=='sses':
> +        return s[0:-4]+'ss'
> +    elif s[-1]=='s':
> +        return s[0:-1]
> +    else:
> +        return s;
>
> Seems frail
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/744#pullrequestreview-11726819>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgDDfipaxnK3oAIrKYYSgZEfWjtYwBjks5rFh1jgaJpZM4LEfIF>
> .
>

alechenninger commented on tools/md2swagger.py:
```diff
@@ -0,0 +1,103 @@
+                if arrayType=='object':
+                    nameStack.append(singular(capital1(field)))
+                    newEntityName=makeEntityName(nameStack)
+                    fieldBlock['items']={'$ref':'#definitions/'+newEntityName}
```
I made some fixes in the internal repo for this you may want to grab
- need #/definitions (missing /)
- The new top level sub entities need type: object and properties: fields

bserdar commented:
Without this, we had UserLegalEntities as a classname.

On Tue, Dec 6, 2016 at 8:20 PM, Dennis Crissman <notifications@github.com>
wrote:

> *@dcrissman* commented on this pull request.
> ------------------------------
>
> In tools/md2swagger.py
> <https://github.com/lightblue-platform/lightblue-core/pull/744>:
>
> > +    elif type=='string':
> +        properties['type']='string'
> +    return properties;
> +
> +def capital1(s):
> +    return s[0].upper()+s[1:]
> +
> +def singular(s):
> +    if s[-3:]=='ies':
> +        return s[0:-3]+'y'
> +    elif s[-4:]=='sses':
> +        return s[0:-4]+'ss'
> +    elif s[-1]=='s':
> +        return s[0:-1]
> +    else:
> +        return s;
>
> Do you have a better idea to convert a plural to a singular?
>
> In fairness, no. My thought was that a word that correctly ends with 's'
> would have problems, but the more I think about it, that may be a fringe
> case anyway.
>
> Could you explain the need for this conversion?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/744>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADgDDUzHOjIgPCpt0cEooQAKpyQuxVjaks5rFiYagaJpZM4LEfIF>
> .
>

bserdar commented:
Added fixes from the internal repo, and added a catch-all for type
properties.

On Tue, Dec 6, 2016 at 8:31 PM, Dennis Crissman <notifications@github.com>
wrote:

> *@dcrissman* commented on this pull request.
> ------------------------------
>
> In tools/md2swagger.py
> <https://github.com/lightblue-platform/lightblue-core/pull/744>:
>
> > +    elif type=='string':
> +        properties['type']='string'
> +    return properties;
> +
> +def capital1(s):
> +    return s[0].upper()+s[1:]
> +
> +def singular(s):
> +    if s[-3:]=='ies':
> +        return s[0:-3]+'y'
> +    elif s[-4:]=='sses':
> +        return s[0:-4]+'ss'
> +    elif s[-1]=='s':
> +        return s[0:-1]
> +    else:
> +        return s;
>
> Without this, we had UserLegalEntities as a classname.
>
> I won't hold this up further on this point. This is certainly good enough
> and we can adjust if/when we find a use case. Thanks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/744>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADgDDWPuq4APfNhTu6QuHUmTpStM2e8zks5rFih0gaJpZM4LEfIF>
> .
>
